server/service/src/processors/assign_requisition_number/assign_requisition_number.rs

```diff
@@ -0,0 +1,84 @@
+use repository::{
+    ChangelogFilter, ChangelogRow, ChangelogTableName, EqualFilter, KeyType, NumberRowType,
+    RequisitionRow, RequisitionRowRepository, RequisitionType,
+};
+
+use crate::{
+    number::next_number,
+    processors::general_processor::{Processor, ProcessorError},
+    service_provider::{ServiceContext, ServiceProvider},
+    sync::ActiveStoresOnSite,
+};
+
+const DESCRIPTION: &str = "Assign requisition number to a response requisition";
+
+pub(crate) struct AssignRequisitionNumber;
+
+impl Processor for AssignRequisitionNumber {
+    fn get_description(&self) -> String {
+        DESCRIPTION.to_string()
+    }
+
+    fn try_process_record(
+        &self,
+        ctx: &ServiceContext,
+        _service_provider: &ServiceProvider,
+        changelog: &ChangelogRow,
+    ) -> Result<Option<String>, ProcessorError> {
+        let repo = RequisitionRowRepository::new(&ctx.connection);
+
+        let requisition =
```

Comment:
I think it is much more straightforward to follow just doing a query for the relevant records in the relevant methods like this. If it ends up duplicated (3 times rule of thumb...) then extract out a fn rather than having a bunch of hard to navigate (via LSP) shared logic somewhere else?